src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -251,7 +251,11 @@ public void Start()
             // the next WalletSyncManager.OrchestrateWalletSync invocation.
             foreach (string walletName in this.WalletRepository.GetWalletNames())
             {
-                this.WalletRepository.RewindWallet(walletName, this.ChainIndexer.Tip);
+                // A wallet ahead of consensus should be truncated.
+                ChainedHeader fork = this.WalletRepository.FindFork(walletName, this.ChainIndexer.Tip);
+
+                if (this.WalletRepository.RewindWallet(walletName, fork).Item1)
```

Comment:
Can you name the tuple variables? Perhaps call this one `RewindExecuted` or something?